plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundReadWrite.java

```diff
@@ -64,46 +113,93 @@ private GroundReadWrite(Configuration configuration) {
+                    graphFactory = (GraphVersionFactory) o;
                 }
                 o = createInstance(nodeFactoryType);
-                if (NodeFactory.class.isAssignableFrom(o.getClass())) {
-                    nodeFactory = (NodeFactory<?>) o;
+                if (NodeVersionFactory.class.isAssignableFrom(o.getClass())) {
+                    nodeVersionFactory = (NodeVersionFactory) o;
                 }
                 o = createInstance(edgeFactoryType);
-                if (EdgeFactory.class.isAssignableFrom(o.getClass())) {
-                    edgeFactory = (EdgeFactory) o;
+                if (EdgeVersionFactory.class.isAssignableFrom(o.getClass())) {
+                    edgeVersionFactory = (EdgeVersionFactory) o;
                 }
             }
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
 
-    private Object createInstance(String clientClass)
-            throws ClassNotFoundException, InstantiationException, IllegalAccessException {
-        Class<?> c = Class.forName(clientClass);
-        Object o = c.newInstance();
-        return o;
+    private void createTestInstances() throws GroundDBException {
+        // TODO move this
+        dbClient = new PostgresClient("127.0.0.1", 5432, "test", "test", "test");
+        PostgresVersionSuccessorFactory succ = new PostgresVersionSuccessorFactory();
+        PostgresVersionHistoryDAGFactory dagFactory = new PostgresVersionHistoryDAGFactory(succ);
+        PostgresItemFactory itemFactory = new PostgresItemFactory(dagFactory);
+        nodeFactory = new PostgresNodeFactory(itemFactory, (PostgresClient) dbClient);
+        VersionFactory vf = new PostgresVersionFactory();
+        ItemFactory iff = new PostgresItemFactory(null);
+        StructureFactory sf = new PostgresStructureFactory((PostgresItemFactory) iff, (PostgresClient) dbClient);
+        StructureVersionFactory svf = new PostgresStructureVersionFactory((PostgresStructureFactory) sf,
+                (PostgresVersionFactory) vf, (PostgresClient) dbClient);
+        RichVersionFactory rf = new PostgresRichVersionFactory((PostgresVersionFactory) vf,
+                (PostgresStructureVersionFactory) svf, null);
+
+        edgeFactory = new PostgresEdgeFactory(itemFactory, (PostgresClient) dbClient);
+        edgeVersionFactory = new PostgresEdgeVersionFactory((PostgresEdgeFactory) edgeFactory,
+                (PostgresRichVersionFactory) rf, (PostgresClient) dbClient);
+        LOG.info("postgresclient " + dbClient.getConnection().toString());
+        nodeVersionFactory = new PostgresNodeVersionFactory((PostgresNodeFactory) nodeFactory,
+                (PostgresRichVersionFactory) rf, (PostgresClient) dbClient);
     }
 
-    public static synchronized void setConf(Configuration conf) {
-        // TODO configuration parameters to initialize metastore properties
+    private Object createInstance(String clientClass)
```

Comment:
I don't understand why this is being implemented this way. Can you guys explain please?